airflow/models/dagrun.py

```diff
@@ -1227,11 +1217,11 @@ def _emit_true_scheduling_delay_stats_for_finished_state(self, finished_tis: lis
         rid of the outliers on the stats side through dashboards tooling.
 
         Note that the stat will only be emitted for scheduler-triggered DAG runs
-        (i.e. when ``external_trigger`` is *False* and ``clear_number`` is equal to 0).
+        (i.e. when ``run_type`` is *MANUAL* and ``clear_number`` is equal to 0).
```

Comment:
```suggestion
        (i.e. when ``run_type`` is not *MANUAL* and ``clear_number`` is equal to 0).
```

Though this brings up another question: How should this function deal with run_type==BACKFILL?

Reply:
How about adding `externally_triggered_type` in `airflow.utils.types` ?
```python
externally_triggered_type: frozenset[DagRunType] = frozenset([DagRunType.BACKFILL_JOB, DagRunType.MANUAL, DagRunType.ASSET_TRIGGERED])
```

Reply:
I think we need to look at what the intent here was, and how that maps on to the new dag run types. My guess is that when "external_trigger" was added, there was either scheduled, or external.

So Manual is clearly not scheduled, but what abouty Backfill runs? How should we treat those. And for that I delegate to @uranusjr and @dstandish *boing flip* 😄

Reply:
After looking deeper, I think `external_trigger is False` is equivalent to `run_type is SCHEDULED` and should fit this case.

Reply:
I dug into the history and yes it seems that `external_trigger` is *False* when `run_type` is set to `BACKFILL_JOB`. Also, `external_trigger` predates both `run_type` and `triggered_by` by a lot, its true definition is quite ambiguous and lost to time, so we can probably take some liberty changing its definition as long as the new implementation makes sense and does not cause bugs.

I think it’d be reasonable enough to define `external_trigger == True` as equivalent to `run_type == "manual"` for refactoring purposes. If this causes bugs, they should be fixed under this definition.